src/main/scala/com/yotpo/metorikku/output/writers/file/HudiOutputWriter.scala

```diff
@@ -312,8 +321,42 @@ class HudiOutputWriter(props: Map[String, Object], hudiOutput: Option[Hudi]) ext
     df
   }
 
+  def hiveSyncManually(dataFrame: DataFrame): Unit ={
+    var df = dataFrame
+    val ss = df.sparkSession
+
+    df = this.hudiOutputProperties.path match {
+      case Some(path) => {
+        ss.read.format("org.apache.hudi").option("path", path).load()
+      }
+    }
+
+    val partitionColumnName = this.hudiOutputProperties.partitionBy match {
+      case Some(partitionBy) => {
+        partitionBy
+      }
+      case None => None
+    }
+
+
+    this.hudiOutputProperties.tableName match {
```

Comment:
So you're missing the notion of creating a new table
You need to support it as well

Reply:
If the table doesn't exist, you need to create the table completely with all the hudi S***
So adding all this for example:
```
PARTITIONED BY (`....` STRING)
ROW FORMAT SERDE 'org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe'
WITH SERDEPROPERTIES (
  'serialization.format' = '1'
)
STORED AS
INPUTFORMAT 'org.apache.hudi.hadoop.HoodieParquetInputFormat'
OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat'
LOCATION 's3://...'

Reply:
If it's merge on read you also need to handle the _rt table

Reply:
We can say we support only non-partitioned or single partition tables, so you can simply use ```createPartitions``` or ```alterPartitions``` here: https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala